driver.go

```diff
@@ -0,0 +1,62 @@
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package storage
+
+import (
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/m3db/m3x/close"
+)
+
+// A SeriesIterator is used to return datapoints from a series
+type SeriesIterator interface {
+	xclose.Closer
+
+	// Next returns true if there is more data in the series
+	Next() bool
+
+	// Value returns the value and timestamp for the current datapoint
+	Value() (float64, time.Time)
+}
+
+// A Connection is a connection to a storage cluster, which can be used to
+// read and write datapoints to that cluster
+type Connection interface {
+	xclose.Closer
+
+	// Read reads datapoints for the given id at the given time range and resolution
+	Read(id string, r Resolution, start, end time.Time) (SeriesIterator, error)
+
+	// Write writes a datapoint for the given id at the given time range
+	Write(id string, r Resolution, t time.Time, value float64) error
+}
+
+// A Driver is used to create connections to a cluster of a given storage class
+type Driver interface {
+	// OpenConnection opens a connection with the provided config
+	OpenConnection(config proto.Message) (Connection, error)
```

Comment:
I dunno, I kind of like having the Connection suffix to be explicit on what is being opened / reconfigured.